website/content/blog/2025-05-30-namespaces-announcement.md

```diff
@@ -0,0 +1,145 @@
+
+```bash
+$ bao secrets enable -ns=tenants/tenant-c/prod -description="Production Secrets" kv
+Success! Enabled the kv secrets engine at: kv/
+```
+
+Now that we have created a new kv secrets engine, it will be listed among our default engines:
+
+```bash
+$ bao secrets list -ns tenants/tenant-c/prod
+Path          Type            Accessor              Description
+----          ----            --------              -----------
+cubbyhole/    ns_cubbyhole    cubbyhole_9a4e44d3    per-token private secret storage
+identity/     ns_identity     identity_61c42408     identity store
+kv/           kv              kv_b993bbec           Production Secrets
+sys/          ns_system       system_ba514ba2       system endpoints used for control, policy and debugging
+```
+
+Let's read and write a secret...
+
+```bash
+# Create the secret within namespace
+$ bao kv put -namespace=tenants/tenant-c/prod -mount=kv auth-token foo=bar
+Success! Data written to: kv/auth-token
+
+# Read it
+$ bao kv get -namespace=tenants/tenant-c/prod -mount=kv auth-token
+=== Data ===
+Key    Value
+---    -----
+foo    bar
+```
+
+3. **Lifecycle Operations**
+
+Over time, tenants will inevitably require various changes and customizations to their namespaces. Moving the administration away from the OpenBao operators means that those changes can be performed on demand and individually by the tenant operator. Consequently, we implemented the associated (and probably expected) lifecycle features such as
+
+* Namespace-aware policies and quotas
+* Moving or renaming secret or auth engines across namespaces
+* Locking and unlocking of namespaces
+
+While we have a lot of plans on how to extend namespace capabilities in the future to make them even more helpful, we still maintain API compatibility with Vault Enterprise to enable a smooth migration path.
+
+## Looking Ahead: Horizontal Scalability
+
+Namespaces will give organisations the possibility to better structure and isolate secret information. However, the introduction of namespaces is just the first step. Our vision includes supporting _lazy loading_ of namespaces and mounts, allowing OpenBao clusters to efficiently serve workloads with many infrequently accessed resources. This will enable even greater scalability and resilience, as nodes will no longer need to load the entire system state at once. 
+
+[Here](https://openbao.org/blog/vision-for-namespaces/) is another article if you want to read more about OpenBaos' scalability efforts. Sounds interesting? Join the [WG-Horizontal-Scalability](https://lists.lfedge.org/g/openbao-dev-wg-scalability) if you want to help building! 
+
+## Get Started
+
+We are planning to release OpenBao 2.3 Beta in the upcoming days. We invite you to explore the documentation, try out the new features, and share your feedback as we continue to evolve OpenBao for secure, scalable, and self-service secrets management.
```

Comment:
I realised we have barely any documentation about namespaces yet. Maybe a call to action to improve namespace documentation would be more adequate?